aptos-move/move-examples/token_objects/ambassador/client/ambassador.ts

```diff
@@ -0,0 +1,171 @@
+    const tokenOwnership = data.data;
+
+    // TODO: `tokenOwnership` can be obtained by the following code when a new SDK version is ready.
+    // const tokenOwnership = await provider.getTokenOwnershipV2(ownerAddr);
+
+    for (const ownership of tokenOwnership.current_token_ownerships_v2) {
+        if (ownership.current_token_data.token_name === tokenName) {
+            return new HexString(ownership.current_token_data.token_data_id);
+        }
+    }
+    console.log(`Token ${tokenName} not found`);
+    process.exit(1);
+}
+
+async function waitForEnter() {
+    return new Promise<void>((resolve, reject) => {
+        const rl = require("readline").createInterface({
+            input: process.stdin,
+            output: process.stdout
+        });
+
+        rl.question('Please press the Enter key to proceed ...\n', () => {
+            rl.close();
+            resolve();
+        });
+    });
+}
+
+class AmbassadorClient extends AptosClient {
+    constructor() {
+        super(NODE_URL);
+    }
+
+    async setAmbassadorLevel(creator: AptosAccount, token: HexString, new_ambassador_level: number | bigint): Promise<string> {
+        const rawTxn = await this.generateTransaction(creator.address(), {
+            function: `${creator.address()}::ambassador::set_ambassador_level`,
+            type_arguments: [],
+            arguments: [token.hex(), new_ambassador_level],
+        });
+
+        const bcsTxn = await this.signTransaction(creator, rawTxn);
+        const pendingTxn = await this.submitTransaction(bcsTxn);
+
+        return pendingTxn.hash;
+    }
+
+    async burn(creator: AptosAccount, token: HexString): Promise<string> {
+        const rawTxn = await this.generateTransaction(creator.address(), {
+            function: `${creator.address()}::ambassador::burn`,
+            type_arguments: [],
+            arguments: [token.hex()],
+        });
+
+        const bcsTxn = await this.signTransaction(creator, rawTxn);
+        const pendingTxn = await this.submitTransaction(bcsTxn);
+
+        return pendingTxn.hash;
+    }
+
+    async mintAmbassadorToken(creator: AptosAccount, description: string, name: string, uri: string, soul_bound_to: HexString): Promise<string> {
+        const rawTxn = await this.generateTransaction(creator.address(), {
+            function: `${creator.address()}::ambassador::mint_ambassador_token`,
+            type_arguments: [],
+            arguments: [description, name, uri, soul_bound_to.hex()],
+        });
+
+        const bcsTxn = await this.signTransaction(creator, rawTxn);
+        const pendingTxn = await this.submitTransaction(bcsTxn);
+
+        return pendingTxn.hash;
+    }
+}
+
+/** run our demo! */
+async function main(): Promise<void> {
+    const client = new AmbassadorClient();
+
+    // Set a test admin account
+    const privateKeyBytes_admin = Uint8Array.from(Buffer.from('f21423f436f7d44c2abd95b5a25323e81fc737040ab17ae8fe40dbf1b1de9e66', 'hex'));
```

Comment:
Can we generate a key instead of hardcoding private keys? I'd avoid having any private key even throwaway ones in the repo

Reply:
I am trying, but it's a bit tricky because of the compilation and publication steps.

Reply:
After creating a random "admin" account, we need to compile and publish the Move package with the account. Especially the compilation step is not easy to automate. For this, I will just follow this way letting the user compile it manually: https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/examples/typescript/your_coin.ts#L105